test/function/prival.cpp

```diff
@@ -134,8 +134,6 @@ namespace {
     result = stumpless_prival_from_string( "syslog.err" );
     EXPECT_EQ( result, -1 );
 
-    EXPECT_ERROR_ID_EQ( STUMPLESS_MEMORY_ALLOCATION_FAILURE );
```

Comment:
If this line causes the test to fail, it seems like the memory allocation failure error is getting overwritten before the return. Since this would be misleading to the caller, please find a way to preserve the error instead. I'm guessing the check and error raising at lines 117 and 118 of `src/prival.c` are the cause, but haven't confirmed by testing it.